data/readreplica.py

```diff
@@ -16,6 +17,25 @@ class ReadOnlyModeException(Exception):
     """
 
 
+_FORCE_MASTER_COUNTER_ATTRIBUTE = "_force_master_nesting"
+
+
+@contextmanager
+def disallow_replica_use(db):
+    """ When used, any queries run under this context manager will hit the master
+        node and be disallowed from using the read replica(s). NOTE: This means if
+        the master node is unavailable, the underlying queries will *fail*.
+    """
+    database = db.obj
+    counter = getattr(database._state, _FORCE_MASTER_COUNTER_ATTRIBUTE, 0)
+    setattr(database._state, _FORCE_MASTER_COUNTER_ATTRIBUTE, counter + 1)
+    try:
+        yield
+    finally:
+        counter = getattr(database._state, _FORCE_MASTER_COUNTER_ATTRIBUTE, 0)
```

Comment:
If we get zero here, we should raise an exception.

Reply:
Done